pages/count.vue

```diff
@@ -0,0 +1,66 @@
+<template>
+  <div class="justify-center flex w-100 flex-col">
+    <div class="bg-white">
+      <p>count: {{ count }}</p>
+      <a @click.prevent="increment" href="#" class="btn btn-primary">+</a>
+      <a @click.prevent="decrement" href="#" class="btn btn-primary">-</a>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  async mounted() {
+    // console.log(this.$fireAuth);
+    try {
+      await this.$store.dispatch("count/bindCountDocument");
+    } catch (e) {
+      console.error(e);
+    }
+  },
+  methods: {
+    async increment() {
+      await this.$store.dispatch("count/increment");
+    },
+    async decrement() {
```

Comment:
why async await these things? Is there some reason you expect them to take time?

why write them with manual methods that call dispatch, and not map actions?


```js
import { mapActions } from 'vuex';

export {
    methods: {
       ...mapActions({
            increment: 'count/increment',
            decrement: 'count/decrement',
       })
    }
}
```

Reply:
for this one i found the pathing, plus how map actions really worked super confusing.

also if you do this, whats the best way to pass params to the action?

Reply:
yeah at first i got really lost with accessing the store, etc, so i wanted to just get used to it

Reply:
lets say increment accepted a number

`this.increment(1)`

```js
increment(state, number) {
    state.count += number;
}```